src/main/java/appeng/me/cache/GridStorageCache.java

```diff
@@ -485,19 +513,45 @@ private void updateBytesInfo() {
                             }
                         }
                     }
+
+                    switch (tc.getCellType(1)) {
+                        case 0 -> this.putItemStackIntoMap(itemCells, stack);
+                        case 1 -> this.putItemStackIntoMap(fluidCells, stack);
+                        case 2 -> this.putEssentiaItemStackIntoMap(essentiaCells, stack);
+                    }
                 }
             }
         } catch (Exception e) {
             // XD Normally won't be here, just normally..
         }
     }
 
+    private void putItemStackIntoMap(final HashMap<ItemStack, Integer> map, final ItemStack stack) {
+        ItemStack newStack = new ItemStack(stack.getItem());
+        if (map.containsKey(newStack)) {
+            map.put(newStack, map.get(newStack) + 1);
+        } else {
+            map.put(newStack, 1);
+        }
+    }
+
+    private void putEssentiaItemStackIntoMap(final HashMap<ItemStack, Integer> map, final ItemStack stack) {
+        ItemStack newStack = new ItemStack(stack.getItem());
+        newStack.setItemDamage(stack.getItemDamage());
+        if (map.containsKey(newStack)) {
+            map.put(newStack, map.get(newStack) + 1);
+        } else {
+            map.put(newStack, 1);
+        }
+    }
```

Comment:
These methods can be combined into one. Also code can be simplified:
```suggestion
    private void putItemStackIntoMap(final ItemStackMap<Integer> map, final ItemStack stack) {
        ItemStack newStack = new ItemStack(stack.getItem(), 1, stack.getItemDamage());
        if (map.computeIfPresent(newStack, (currentStack, count) -> count + 1) == null) {
            map.put(newStack, 1);
        }
    }
```

Reply:
thats cool, thx